external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -1779,6 +1775,21 @@ fn parse_binop_exp(context: &mut Context, lhs: Exp, min_prec: u32) -> Result<Exp
         context.tokens.advance()?;
         let op_end_loc = context.tokens.previous_end_loc();
 
+        if op_token == Tok::As {
+            let ty = {
+                // numeric types don't have type arguments, so don't attempt to parse '<' as a
+                // type arg
+                let tn = parse_name_access_chain(context, || "a type name")?;
+                sp(tn.loc, Type_::Apply(Box::new(tn), vec![]))
+            };
+            let start_loc = result.loc.start() as usize;
+            let end_loc = context.tokens.previous_end_loc();
+            let e_ = Exp_::Cast(Box::new(result), ty);
+            result = spanned(context.tokens.file_hash(), start_loc, end_loc, e_);
+            next_tok_prec = get_precedence(context.tokens.peek());
+            continue;
```

Comment:
I'm just amazed this whole think worked. And don't really understand what I wrote. So an extra set of eyes here would be helpful